prusti-contracts/prusti-specs/src/specifications/preparser.rs

```diff
@@ -905,14 +946,16 @@ impl RustOp {
 }
 
 fn join_spans(s1: Span, s2: Span) -> Span {
-    // Tests don't run in the proc macro context
-    let is_proc_macro = std::panic::catch_unwind(|| s1.unwrap()).is_ok();
-    if is_proc_macro {
-        // This works even when compiled with stable
-        s1.unwrap().join(s2.unwrap()).expect("Failed to join spans!").into()
-    } else {
+    // Tests don't run in the proc macro context, so this gets a little funky for them
```

Comment:
@JonasAlaif I think I recall we talked about this at some point. Does this `cfg` solution look good to you? Did you try this before the `catch_unwind` one?

Reply:
I guess it's late now anyway, but this solution looks better yeah